IBM/CAST
# Issues/770 csm allocation step query details segmentation fault

# Purpose
To fix issue https://github.com/IBM/CAST/issues/770 but not having the cmd line api seg fault. 

# Approach
I decided to go with: 
- don't leave out any data from the user.
- show them all records that matched the query, even if there are multiple steps with the same step id in that allocation. which will probably never happen in production.

# Origin
https://github.com/IBM/CAST/issues/770

@fpizzano  found an issue. very rare edge case. will probably never happen in production. where if there are multiple steps with a matching id under the same allocation, then allocation step query details cmd line interface will segfault. 

# How to Test
_This section will help the regression tester._
1. manually create the situation in a test database.
2. run the api.
3. see if the seg fault still happens. 
4. you can look at https://github.com/IBM/CAST/issues/770 for an example

# Screenshots
screen shots of terminal output are in https://github.com/IBM/CAST/issues/770
